### PR TITLE
Add recommendationLog service

### PR DESCRIPTION
**Pull request by jordanreedie**
## Description
Add the recommendationLog service to support logging of the recommendations that are generated in mattermost-webapp.

## Motivation and context
We want to be able to track how often a user visits their dashboard & the status of the recommendations that are generated. We can use this data for more in-depth analytics about how our users use & are influenced by the dashboard

## How has this been tested?
Tested locally

## Types of changes

- ✅ New feature (non-breaking change which adds functionality)

## Checklist:

- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ I have added tests to cover my changes.

### CODE REVIEW

**Comment by mlippert** [comment]
commits updating dependencies, removing obsolete reports code and fixing some lint errors were moved into #17 and merged. I've cherry-picked the commits related to creating the recommendationLog service on top of that merged code and force pushed. I'll update the summary and description of this PR.

File: src/services/recommendationLog/recommendationLog-model.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+const Schema = mongoose.Schema
+
+const recommendationLogSchema = new Schema({
+  participantId: String,
+  timestamp: { type: Date, 'default': Date.now },
+  recommendations: [{
+    name: String,
+    isComplete: Boolean,
+  }]
+})
```
Comment (mlippert) [inline src/services/recommendationLog/recommendationLog-model.js]: I was expecting better documentation of this recommendationLog schema from @jaedoucette and @jordanreedie .
What does a recommendationLog document (ie record) represent? Whose participantId is this, what are the recommandations in that array, what are the values (or at least some examples of what is in `name`, what does it mean when `isComplete` is true vs false. Even the timestamp is recording when the recommendationLog document is created (I'm assuming it shouldn't be changed after that) but what time is it that requires it to be recorded?

File: src/services/meeting/hooks/participants-event-hook.js [hunk 2]
```diff
@@ -15,7 +15,7 @@ module.exports = function (hook) {
         timestamp: new Date()
       })
       .then((participantEvent) => {
-        winston.log('info', 'created participantEvent!', participantEvent)
+        winston.log('info', 'created participantEvent!')
```
Comment (mlippert) [inline src/services/meeting/hooks/participants-event-hook.js]: Why this change?

Comment (jordanreedie) [reply src/services/meeting/hooks/participants-event-hook.js]: Hmm I'm not sure actually, I seem to recall removing a log that was resulting in 100+ lines being printed but after testing it just now I don't think this is that one

Comment (mlippert) [reply src/services/meeting/hooks/participants-event-hook.js]: Should I put it back? If it isn't super large (and I agree, I don't think this one would be), it seems useful log message information,

Comment (mlippert) [reply src/services/meeting/hooks/participants-event-hook.js]: I put it back.

**Comment by mlippert** [comment]
Squashed the commits, and added 1 new one changing the test from using the done callback for testing the returned promise to using async/await. This "fixes" the lint errors in the file and seems like a good style to use for testing promises.
I got the idea from: https://wietse.loves.engineering/testing-promises-with-mocha-90df8b7d2e35

**Comment by mlippert** [comment]
**Release** [0.4.0-dev.14](https://github.com/rifflearning/riff-server/releases/tag/0.4.0-dev.14)
